### PR TITLE
[MU4] Loop cursor

### PR DESCRIPTION
**Pull request by RomanPudashkin**
- Implemented loop cursor
- Made a bunch of minor code-style fixes

https://user-images.githubusercontent.com/67867444/107398112-fb4cf580-6b07-11eb-89c4-55e8d061a3ba.mp4

### CODE REVIEW

File: src/framework/audio/internal/rpc/rpcsequencer.h [hunk 1]
```diff
@@ -47,11 +47,10 @@ class RpcSequencer : public ISequencer
     void stop() override;
     void seek(uint64_t position) override;
     void rewind() override;
-    void setLoop(uint64_t fromMiliSeconds, uint64_t toMiliSeconds) override;
+    void setLoop(uint64_t fromMilliSeconds, uint64_t toMilliSeconds) override;
```
Comment (Eism) [inline src/framework/audio/internal/rpc/rpcsequencer.h:50]: seconds to lowercase, please

File: src/notation/internal/notationplayback.h [hunk 2]
```diff
@@ -47,15 +46,21 @@ class NotationPlayback : public INotationPlayback, public async::Asyncable
...
     void setPlayPositionTick(int tick) override;
-    bool setPlayPositionByElement(const Element* e) override;
+    bool setPlayPositionByElement(const Element* element) override;
     async::Channel<int> playPositionTickChanged() const override;
 
-    midi::MidiData playElementMidiData(const Element* e) const override;
+    midi::MidiData playElementMidiData(const Element* element) const override;
+
+    void addLoopInToSelectedNote() override;
+    void addLoopOutToSelectedNote() override;
```
Comment (Eism) [inline src/notation/internal/notationplayback.h:59]: we can combine addLoopInToSelectedNote and addLoopOutToSelectedNote to one with input parameter loopType

File: src/notation/view/loopcursor.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
```
Comment (Eism) [inline src/notation/view/loopcursor.cpp:5]: 2021 :)

File: src/notation/view/loopcursor.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+{
+    if (!m_visible || !m_style) {
+        return;
+    }
+
+    QPolygonF triangle(3);
+
+    qreal x = m_rect.left();
+    qreal y = m_rect.top();
+    qreal h = m_style->styleValue(StyleId::spatium).toDouble() * 2;
```
Comment (Eism) [inline src/notation/view/loopcursor.cpp:54]: I think the height should also depend on the staff, we can have a different number of lines on the staff

File: src/notation/view/loopcursor.h [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    LoopCursor(LoopCursorType type);
+
+    void setRect(const QRect& rect);
+    void setVisible(bool visible);
+    void setStyle(INotationStylePtr style);
+
+    void paint(QPainter* painter);
+
+private:
+    LoopCursorType m_type;
```
Comment (Eism) [inline src/notation/view/loopcursor.h:42]: please, add the initialized value

File: src/framework/audio/internal/rpc/rpcsequencer.h (hunk 1 above)
Comment (RomanPudashkin) [reply src/framework/audio/internal/rpc/rpcsequencer.h]: fixed

File: src/notation/internal/notationplayback.h (hunk 2 above)
Comment (RomanPudashkin) [reply src/notation/internal/notationplayback.h]: fixed

File: src/notation/view/loopcursor.cpp (hunk 3 above)
Comment (RomanPudashkin) [reply src/notation/view/loopcursor.cpp]: fixed

File: src/notation/view/loopcursor.h (hunk 5 above)
Comment (RomanPudashkin) [reply src/notation/view/loopcursor.h]: fixed

File: src/notation/view/loopcursor.cpp (hunk 4 above)
Comment (RomanPudashkin) [reply src/notation/view/loopcursor.cpp]: This is transferred implementation from MU3

**Review by Eism — APPROVED** [review]

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**